denops/@denops-private/integration_test.ts

```diff
@@ -0,0 +1,27 @@
+import { assertMatch } from "https://deno.land/std@0.210.0/assert/mod.ts";
+import { delay } from "https://deno.land/std@0.210.0/async/delay.ts";
+import { stub } from "https://deno.land/std@0.210.0/testing/mock.ts";
+import { withDenops } from "https://deno.land/x/denops_test@v1.8.0/mod.ts";
+
+for (const mode of ["vim", "nvim"] as const) {
+  Deno.test(`denops#server#close() should fires DenopsClosed (${mode})`, async () => {
+    const console_error = stub(console, "error");
+    try {
+      await withDenops(mode, async (denops) => {
+        await denops.cmd(
+          "autocmd User DenopsClosed echomsg 'DenopsClosed fired!'",
+        );
+        // NOTE: It does not resolved this call, so no await.
+        /* no await */ denops.call("denops#server#close");
+        // NOTE: Delay for event fired.
+        await delay(100);
+      }, { verbose: true });
+      // NOTE: The output may be split into random points, so join them all.
+      const outputs = console_error.calls
+        .flatMap((c) => c.args).map((a) => String(a)).join("");
+      assertMatch(outputs, /DenopsClosed fired!/);
+    } finally {
+      console_error.restore();
+    }
+  });
+}
```

Comment:
Let's add integration tests under `/tests/denops` directory so that we can distinguish with unit tests

For example

```
 denops.vim/
 │  .git/
 │  .github/
 │  autoload/
 │  denops/
 │  doc/
 │  plugin/
 │  tests/
 │    themis/
 │   └  purpose_of_tests.vim
 │    denops/
 │   └  purpose_of_tests.ts
 │  .gitignore
 │  .gitmessage
 │  LICENSE
 │  README.md
 │  deno.jsonc
 └  deno.lock
```